internal/services/v1/watch.go

```diff
@@ -0,0 +1,106 @@
+package v1
+
+import (
+	"errors"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/authzed/spicedb/internal/datastore"
+	"github.com/authzed/spicedb/internal/namespace"
+	"github.com/authzed/spicedb/internal/services/shared"
+	"github.com/authzed/spicedb/pkg/zedtoken"
+	grpcvalidate "github.com/grpc-ecosystem/go-grpc-middleware/validator"
+	"github.com/shopspring/decimal"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type watchServer struct {
+	v1.UnimplementedWatchServiceServer
+	shared.WithStreamServiceSpecificInterceptor
+
+	ds  datastore.Datastore
+	nsm namespace.Manager
+}
+
+// NewWatchServer creates an instance of the watch server.
+func NewWatchServer(ds datastore.Datastore, nsm namespace.Manager) v1.WatchServiceServer {
+	s := &watchServer{
+		ds: ds,
+		WithStreamServiceSpecificInterceptor: shared.WithStreamServiceSpecificInterceptor{
+			Stream: grpcvalidate.StreamServerInterceptor(),
+		},
+	}
+	return s
+}
+
+func (ws *watchServer) Watch(req *v1.WatchRequest, stream v1.WatchService_WatchServer) error {
+	objectTypesMap := make(map[string]struct{})
+	for _, objectType := range req.GetObjectTypes() {
+		objectTypesMap[objectType] = struct{}{}
+	}
+	filter := objectTypeFilter{objectTypes: objectTypesMap}
```

Comment:
Why the new filter type vs just sending the map?

Reply:
No particular reason other than it was similar to the other pattern used in v0's implementation. I changed it though. I like passing it in to a function better anyways.